third_party/pyth/price-service/src/rest.ts

```diff
@@ -113,6 +120,55 @@ export class RestAPI {
       "api/latest_vaas?ids[]=<price_feed_id>&ids[]=<price_feed_id_2>&.."
     );
 
+    const getVaaInputSchema: schema = {
+      query: Joi.object({
+        id: Joi.string()
+          .regex(/^(0x)?[a-f0-9]{64}$/)
+          .required(),
+        // publishTime is of UNIX timestamp type
+        publish_time: Joi.number().required(),
+      }).required(),
+    };
+    app.get(
+      "/api/get_vaa",
+      validate(getVaaInputSchema),
+      (req: Request, res: Response) => {
+        const priceFeedId = req.query.id as string;
+        const publishTime = Number(req.query.publish_time as string);
+        const vaa = this.priceFeedVaaInfo.getVaa(priceFeedId, publishTime);
+        // if publishTime is older than cache ttl or vaa is not found, fetch from db
+        if (
+          publishTime <
+            Math.floor(Date.now() / 1000) -
+              PYTH_WRITER_INSERT_LATENCY_SECONDS ||
```

Comment:
I don't think injecting pyth writer logic be good here. The check seems unnecessary to me.